unit-tests/unit-tests-live.cpp

```diff
@@ -5915,3 +5915,114 @@ TEST_CASE("l500_presets_set_preset", "[live]")
        
     }
 }
+
+TEST_CASE("D55 frame drops", "[live]")
+{
+    // Require at least one device to be plugged in
+    rs2::context ctx;
+
+    if (make_context(SECTION_FROM_TEST_NAME, &ctx))
+    {
+        auto list = ctx.query_devices();
+        REQUIRE(list.size());
+        REQUIRE(list.front().supports(RS2_CAMERA_INFO_PRODUCT_LINE));
+        auto dev = list.front();
+
+        // 424x240
+        // 480x270
+        std::vector<std::pair<int, int>> resolutions = { { 424,240 }, { 480, 270 } };
+        for (auto& resolution : resolutions)
+        {
+            int width = resolution.first;
+            int height = resolution.second;
+            std::cout << "RGB resolution: " << width << "x" << height << std::endl;
+            std::mutex mutex;
+            std::mutex mutex_process;
+            std::vector<unsigned long long> rgb_frames_num;
+            rs2::sensor rgb_sensor;
+            std::vector<rs2::stream_profile > rgb_stream_profiles;
+            auto res = configure_all_supported_streams(dev, width, height, 90);
+            for (auto& s : res.first)
+            {
+                s.close();
+            }
+            for (auto& profile : res.second)
+            {
+                auto fps = profile.fps;
+                for (auto& s : res.first)
+                {
+                    auto stream_profiles = s.get_stream_profiles();
+                    for (auto& sp : stream_profiles)
+                    {
+                        if (!(sp.stream_type() == profile.stream && sp.fps() == fps && sp.stream_index() == profile.index && sp.format() == profile.format)) continue;
+                        auto vid = sp.as<rs2::video_stream_profile>();
+                        auto h = vid.height();
+                        auto w = vid.width();
+                        if (!(w == profile.width && h == profile.height)) continue;
+                        if (sp.stream_type() == RS2_STREAM_COLOR)
+                        {
+                            rgb_stream_profiles.push_back(sp);
+                            rgb_sensor = s;
+                        }
+                    }
+                }
+            }
```

Comment:
It is not clear why to repeat the content of `configure_all_supported_streams`.
Either use it or configure locally

Reply:
Not repeated, I filtered out irrelevant profiles (depth, ir) and assigned RGB sensor